examples/link/link.html

```diff
@@ -60,49 +62,89 @@
             console.log(convertToRaw(content));
           };
 
-          this.addLink = this._addLink.bind(this);
+          this.promptForLink = this._promptForLink.bind(this);
+          this.onURLChange = (e) => this.setState({urlValue: e.target.value});
+          this.confirmLink = this._confirmLink.bind(this);
           this.removeLink = this._removeLink.bind(this);
         }
 
-        _addLink(e) {
+        _promptForLink(e) {
+          e.preventDefault();
           const {editorState} = this.state;
           const selection = editorState.getSelection();
-          if (selection.isCollapsed()) {
-            return;
+          if (!selection.isCollapsed()) {
+            this.setState({
+              showURLInput: true,
+              urlValue: '',
+            }, () => {
+              setTimeout(() => this.refs.url.focus(), 0);
```

Comment:
@hellendag any chance you remember why you used `setTimeout` here? :)

I ask because I'm having issues with my Editor's focus state getting out of sync with the parent component's state.